### ISSUE TITLE
[REVIEW]: Qiskit VSCode: a tool to enhance the quantum computing programming experience

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1188]
**Submitting author:** @cbjuan (<a href="http://orcid.org/0000-0003-2045-8329">Juan Cruz-Benito</a>)
**Repository:** <a href="https://github.com/Qiskit/qiskit-vscode/" target ="_blank">https://github.com/Qiskit/qiskit-vscode/</a>
**Version:** v0.4.1
**Editor:** @danielskatz
**Reviewer:** @malmaud, @naberek
**Archive:** Pending

## Status

[![status](http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219/status.svg)](http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219"><img src="http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219/status.svg)](http://joss.theoj.org/papers/7687665c24744f6cc3aa30666a70b219)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Qiskit/qiskit-vscode/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@malmaud & @naberek, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.theoj.org/about#reviewer_guidelines. Any questions/concerns please let @danielskatz know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Qiskit/qiskit-vscode/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (v0.4.1)?
- [x] **Authorship:** Has the submitting author (@cbjuan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Qiskit/qiskit-vscode/">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Version:** Does the release version given match the GitHub release (v0.4.1)?
- [ ] **Authorship:** Has the submitting author (@cbjuan) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **References:** Do all archival references that should have a DOI list one (e.g., papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @malmaud, it looks like you're currently assigned as the reviewer for this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01188/joss.01188/10.21105.joss.01188.pdf)

**Comment by danielskatz** [comment]
👋 @malmaud & @naberek - thanks for agreeing to perform this review.  See the comments above, and let me know if you have any problems or concerns

**Comment by malmaud** [comment]
I'm currently blocked since the VScode extension appears to automatically install the latest version of qiskit terra (0.7), and yet only works with 0.6. 

I'm not sure how to manually downgrade since the qiskit system is split into several Python packages and I don't know what version each of them needs to be to result in a usable configuration (and I don't really have the time to figure it out). Users shouldn't be expected to manually use pip to downgrade to a configuration of qiskit packages that works when they first install this extension and I don't think this can be accepted until the installation is seamless.  

A Github issue does indicate an effort is underway to support terra to 0.7 so I would suggest suspending this review until that gets completed.

**Comment by danielskatz** [comment]
I'm willing to wait.

@cbjuan - please confirm this makes sense for you, and let us know about when you expect this issue to be resolved.

**Comment by cbjuan** [comment]
Hi @malmaud, @danielskatz 

We have almost finished the support of the new version of Qiskit (v0.7) https://github.com/Qiskit/qiskit-vscode/pull/61. We hope to release the new version tomorrow. Let's wait for that to make easier the review. Thanks for your support.

**Comment by cbjuan** [comment]
Sorry for the delay, but we finally released the Qiskit-vscode version v0.5.0 today https://github.com/Qiskit/qiskit-vscode/releases/tag/v0.5.0. You can use it for the review.

https://marketplace.visualstudio.com/items?itemName=qiskit.qiskit-vscode

**Comment by malmaud** [comment]
Blocked by https://github.com/Qiskit/qiskit-vscode/issues/68

**Comment by danielskatz** [comment]
@cbjuan - any thoughts on @malmaud's comment above?

**Comment by danielskatz** [comment]
@naberek - can I get an update from you?

**Comment by cbjuan** [comment]
Hi @danielskatz. We are trying to fix it. It seems to be an error related to the new version of VSCode. We hope to have a solution ASAP (not only for this review but also for the other users reporting the error).

**Comment by danielskatz** [comment]
Hi - I'm going to be on vacation for 2 1/2 weeks, so I'm shifting the editor of this submission to be @arfon, during that period - thanks @arfon

**Comment by danielskatz** [comment]
@whedon assign @arfon as editor

**Comment by cbjuan** [comment]
We solved the issue that was blocking the review in the new version [0.5.2](https://github.com/Qiskit/qiskit-vscode/releases/tag/v0.5.2).

Hope it allows the review as expected.

**Comment by arfon** [comment]
@malmaud, @naberek - seems like @cbjuan has fixed the installation issues. Could you please try again?

**Comment by malmaud** [comment]
Ya, I'm currently traveling but will try again in a few days.

**Comment by malmaud** [comment]
Getting https://github.com/Qiskit/qiskit-vscode/issues/81 at the moment.  I'm guessing some kind of version mismatch in my installed qiskit packages, although I just followed the installation prompts for qiskit that the extension itself offered with VSCode.

**Comment by cbjuan** [comment]
Thanks @malmaud 

It should be solved in [v0.5.4](https://github.com/Qiskit/qiskit-vscode/releases/tag/v0.5.4). Please update your extension in VSCode.

**Comment by malmaud** [comment]
That worked. Running into https://github.com/Qiskit/qiskit-vscode/issues/84 now though.

At this point, I can't really be involved in more rounds of debugging. I'd be happy to either write a review now, given how everything stands, or review at some later point after things have perhaps matured a bit.

**Comment by arfon** [comment]
@cbjuan - given it's not been possible for @malmaud to even get to the point of installing your software yet, I think we should pause this review until you have taken some additional steps to verify that the installation process is a smooth one for new users. 

For example, if you have access to a second machine, perhaps try following your own installation instructions to see how far you can get? Alternatively, perhaps a colleague could try installing the package too?

JOSS reviewers can only be expected to do so much debugging of installation challenges and I think we've reached this limit.

**Comment by cbjuan** [comment]
@malmaud thanks for your input. In this case, we are not aware of this behavior, and we want to test it in depth. 
@arfon at the moment @malmaud can install the extension and use it (at least partially). In any case, thanks for the review and we agree it is the moment to pause it. We are sorry for the review process; several changes in the VSCode APIs led us to provide an unstable version for the review with the new versions of the IDE.

We will keep in touch in the future.

**Comment by danielskatz** [comment]
@whedon assign @danielskatz as editor

**Comment by danielskatz** [comment]
Thanks @arfon for the help while I was on holiday

**Comment by danielskatz** [comment]
@cbjuan - How are things going?
If there's been progress, or will be soon, let us know.
If not, I wonder if we should withdraw this submission with the idea that a new submission could be made later.

**Comment by danielskatz** [comment]
I'll give this another week to get an update from @cbjuan - if not, I'll withdraw it, again with the idea that a new submission can be made later once the installation process is smoother.

**Comment by cbjuan** [comment]
Hi @danielskatz. We did not make any important improvement since the last review. So, honestly, I think that would be better to withdraw the paper in its current status to later re-submit it if we achieve the desired quality standard.

Thanks for the workflow and the reviews to all of you.

**Comment by danielskatz** [comment]
Thanks @cbjuan, and thanks @malmaud for your work in this review, which even if not complete is helpful for the future of the software.

I'll now withdraw this submission
